static/app/utils/useUpdateOrganization.tsx

```diff
@@ -0,0 +1,74 @@
+import OrganizationStore from 'sentry/stores/organizationStore';
+import type {Organization} from 'sentry/types/organization';
+import {
+  fetchMutation,
+  setApiQueryData,
+  useMutation,
+  useQueryClient,
+  type ApiQueryKey,
+} from 'sentry/utils/queryClient';
+
+interface Variables extends Partial<Organization> {}
+
+type Context =
+  | {
+      previousOrganization: Organization;
+      error?: never;
+    }
+  | {
+      error: Error;
+      previousOrganization?: never;
+    };
+
+function makeDetailedOrganizationQueryKey(organization: Organization): ApiQueryKey {
+  return [`/organizations/${organization.slug}/`];
+}
+
+export function useUpdateOrganization(organization: Organization) {
+  const queryClient = useQueryClient();
+
+  const queryKey = makeDetailedOrganizationQueryKey(organization);
+
+  return useMutation<Organization, Error, Variables, Context>({
```

Comment:
let’s please try to avoid this, it should be inferred from usages:


```suggestion
  return useMutation({
```